In [1]:
import pandas as pd  
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from collections import Counter

**Data has been partially cleaned in Excel:**
- Removed blank 'Code X' columns
- Combined testing date columns into one, as all testing occurs at the same time.
- Removed 3 rows with no scoring data (2 are likely duplicates anyway)

In [2]:
evals_raw= pd.DataFrame(pd.read_excel('../data/bdi3_all_evals_clean.xlsx'))
evals_raw.head()

Child ID Gender Date of Birth Location - Sub Level 1  \
0    44879      F    2022-09-27              Southwest   
1    47795      F    2020-12-23      Greater Nashville   
2    54340      M    2021-06-22        First Tennessee   
3    54344      F    2022-03-07    Southeast Tennessee   
4    54515      M    2021-11-04          Memphis Delta   

                  Program Label Date of Testing  \
0  BDI-3 Eligibility Evaluation      2023-06-23   
1  BDI-3 Eligibility Evaluation      2023-03-21   
2  BDI-3 Eligibility Evaluation      2023-04-06   
3  BDI-3 Eligibility Evaluation      2023-02-23   
4  BDI-3 Eligibility Evaluation      2023-02-24   

   Adaptive Sum of Scaled Scores  Adaptive Developmental Quotient  \
0                            7.0                             85.0   
1                           16.0                             88.0   
2                            8.0                             90.0   
3                           13.0                            115.0   
4                            5.0                             75.0   

  Adaptive Percentile Rank Adaptive 95% Confidence Interval  ...  \
0                       16                            77-97  ...   
1                       21                            82-96  ...   
2                       25                           81-101  ...   
3                       84                          103-123  ...   
4                        5                            68-88  ...   

  Cognitive-Perception and Concepts RS Cognitive-Perception and Concepts SS  \
0                                  9.0                                 13.0   
1                                 11.0                                  7.0   
2                                 10.0                                  7.0   
3                                  9.0                                 10.0   
4                                 10.0                                 10.0   

   Cognitive-Perception and Concepts PR Cognitive-Perception and Concepts AE  \
0                                    84                                   12   
1                                    16                                   18   
2                                    16                                   15   
3                                    50                                   12   
4                                    50                                   15   

   Cognitive-Perception and Concepts RDI  \
0                                  98/90   
1                                  29/90   
2                                  37/90   
3                                  92/90   
4                                  88/90   

   Cognitive-Perception and Concepts CSS  \
0                                  411.0   
1                                  437.0   
2                                  426.0   
3                                  411.0   
4                                  426.0   

  Cognitive-Perception and Concepts CSS 90%  \
0                                 392 - 431   
1                                 421 - 453   
2                                 408 - 444   
3                                 392 - 431   
4                                 408 - 444   

  Cognitive-Perception and Concepts Z-Score  \
0                                         1   
1                                    – 1.00   
2                                    – 1.00   
3                                         0   
4                                         0   

  Cognitive-Perception and Concepts T-Score  \
0                                      60.0   
1                                      40.0   
2                                      40.0   
3                                      50.0   
4                                      50.0   

  Cognitive-Perception and Concepts NCE  
0                                    71  
1                                    29  
2                                    29  
3                  

In [8]:
evals_raw['is_duplicate'] = evals_raw.duplicated(subset=['Child ID', 'Date of Testing'])
evals = evals_raw[evals_raw['is_duplicate']==False]
evals = evals.drop('is_duplicate', axis=1)
evals.info()
evals.describe()

<class 'pandas.core.frame.DataFrame'>
Index: 19435 entries, 0 to 23943
Columns: 197 entries, Child ID to Cognitive-Perception and Concepts NCE
dtypes: datetime64[ns](2), float64(69), object(126)
memory usage: 29.4+ MB


Date of Birth                Date of Testing  \
count                          19435                          19435   
mean   2021-08-25 17:37:18.538718976  2023-06-28 20:48:37.046565376   
min              2019-10-15 00:00:00            2022-12-08 00:00:00   
25%              2020-12-11 00:00:00            2023-04-21 00:00:00   
50%              2021-07-21 00:00:00            2023-07-03 00:00:00   
75%              2022-04-12 00:00:00            2023-09-08 12:00:00   
max              2023-10-30 00:00:00            2023-11-15 00:00:00   
std                              NaN                            NaN   

       Adaptive Sum of Scaled Scores  Adaptive Developmental Quotient  \
count                   19430.000000                     19430.000000   
mean                       11.413639                        87.425888   
min                         1.000000                        46.000000   
25%                         7.000000                        76.000000   
50%                        11.000000                        88.000000   
75%                        15.000000                       100.000000   
max                        36.000000                       148.000000   
std                         5.821864                        17.435036   

       Adaptive T-Score  Social-Emotional Sum of Scaled Scores  \
count      19430.000000                           19420.000000   
mean          41.618682                              22.381874   
min           14.000000                               2.000000   
25%           34.000000                              17.000000   
50%           42.000000                              22.000000   
75%           50.000000                              28.000000   
max           82.000000                              53.000000   
std           11.610373                               7.519883   

       Social-Emotional Developmental Quotient  Social-Emotional T-Score  \
count                             19420.000000              19420.000000   
mean                                 94.463131                 46.309784   
min                                  46.000000                 14.000000   
25%                                  84.000000                 39.000000   
50%                                  96.000000                 47.000000   
75%                                 106.000000                 54.000000   
max                                 154.000000                 86.000000   
std                                  16.702164                 11.137009   

       Communication Sum of Scaled Scores  \
count                        19410.000000   
mean                            13.197888   
min                              2.000000   
25%                              8.000000   
50%                             13.000000   
75%                             18.000000   
max                             38.000000   
std                              6.723639   

       Communication Developmental Quotient  ...  \
count                          19410.000000  ...   
mean                              79.593663  ...   
min                               46.000000  ...   
25%                               64.000000  ...   
50%                               79.000000  ...   
75%                               94.000000  ...   
max                              154.000000  ...   
std                               20.170917  ...   

       Cognitive-Attention and Memory CSS  \
count                        19423.000000   
mean                           461.912269   
min                            366.000000   
25%                            452.000000   
50%                            472.000000   
75%                            482.000000   
max                            544.000000   
std                             32.561691   

       Cognitive-Attention and Memory T-Score  \
count                            19423.000000   
mean                                44.704577   
min              

There are 71 of 197 columns formatted in a way that allows for calculations.<br>Not all columns of scoring data are cleanly formatted; need to convert form type 'object' to a number format.

In [9]:
evals['Adaptive Percentile Rank']

0        16
1        21
2        25
3        84
4         5
         ..
23939    84
23940    84
23941    58
23942    21
23943    63
Name: Adaptive Percentile Rank, Length: 19435, dtype: object

In [4]:
evals.value_counts('Program Label')

Program Label
BDI-3 Eligibility Evaluation          12344
BDI-3 Annual Evaluation                6888
BDI-3 Milestone or Exit Evaluation      203
Name: count, dtype: int64

In [5]:
evals.value_counts('Location - Sub Level 1')

Location - Sub Level 1
Greater Nashville      3405
East Tennessee         3337
Memphis Delta          2712
South Central          2555
Upper Cumberland       2018
First Tennessee        1689
Southeast Tennessee    1579
Southwest              1099
Northwest               954
Name: count, dtype: int64

In [6]:
evals.value_counts('Gender')

Gender
M    11921
F     7223
U      291
Name: count, dtype: int64